src/core/public/chrome/ui/header/header_search_bar.tsx

```diff
@@ -0,0 +1,259 @@
+ */
+export const SAVED_OBJECTS_SYMBOL = '@';
+export const COMMANDS_SYMBOL = '>';
+
+export const SearchCommandFilters = {
+  [SearchObjectTypes.PAGES]: (value: string) => {
+    return {
+      match: !value.startsWith(SAVED_OBJECTS_SYMBOL) && !value.startsWith(COMMANDS_SYMBOL),
+      searchValue: value,
+    };
+  },
+  [SearchObjectTypes.SAVED_OBJECTS]: (value: string) => {
+    return {
+      match: value.startsWith(SAVED_OBJECTS_SYMBOL),
+      searchValue: value.replace(SAVED_OBJECTS_SYMBOL, '').trim(),
+    };
+  },
+};
+
+export const HeaderSearchBarIcon = ({ globalSearchCommands }: Props) => {
+  const [isSearchPopoverOpen, setIsSearchPopoverOpen] = useState(false);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  return (
+    <EuiPopover
+      panelPaddingSize="s"
+      anchorPosition="downCenter"
+      repositionOnScroll={true}
+      isOpen={isSearchPopoverOpen}
+      closePopover={() => {
+        setIsSearchPopoverOpen(false);
+      }}
+      button={
+        <EuiToolTip
+          content={i18n.translate('core.globalSearch.icon.toolTip', {
+            defaultMessage: 'Search',
+          })}
+        >
+          <EuiButtonIcon
+            aria-label="search"
+            iconType="search"
+            color="text"
+            buttonRef={buttonRef}
+            data-test-subj="search-icon"
+            onClick={() => {
+              setIsSearchPopoverOpen(!isSearchPopoverOpen);
+              // remove focus from the button to dismiss the tooltip
+              buttonRef.current?.blur();
+            }}
+          />
+        </EuiToolTip>
+      }
+    >
+      <EuiPanel
+        hasBorder={false}
+        hasShadow={false}
+        paddingSize="none"
+        style={{ minHeight: '300px', minWidth: '400px' }}
+      >
+        <HeaderSearchBar
+          globalSearchCommands={globalSearchCommands}
+          panel
+          onSearchResultClick={() => {
+            setIsSearchPopoverOpen(false);
+            buttonRef.current?.blur();
+          }}
+        />
+      </EuiPanel>
+    </EuiPopover>
+  );
+};
+
+export const HeaderSearchBar = ({ globalSearchCommands, panel, onSearchResultClick }: Props) => {
+  const [results, setResults] = useState([] as React.JSX.Element[]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [panelWidth, setPanelWidth] = useState(0);
+  const inputElRef = useRef<HTMLElement | null>();
+  const inputRef = (node: HTMLElement | null) => (inputElRef.current = node);
+
+  const closePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const pagesSection = (items: ReactNode[]) => {
+    return (
+      <EuiFlexGroup direction="column" gutterSize="xs">
+        <EuiFlexItem>
+          <EuiTitle size="s">
+            <EuiText size="xs" color="subdued">
+              {i18n.translate('core.globalSearch.pageSection.title', { defaultMessage: 'Pages' })}
```

Comment:
> This is again just for discussion.
> 
> It seems we need a specific `pagesSection` based on the `type` just to render the title, can we make it so that it's part of the command interface? Like:
> 
> ```
> {
>   id: 'workspacePagesSearchCommand',
>   category: 'Pages',
> }
> ```
> 
> Then for all the items returns by this command, we display it under the `Pages` title. Seems then we don't need `type: SearchObjectTypes;`

Yeah, the section render could be a general component/method, the only difference is the section header. Refactor it to accept a section header props to make it reusable.

Reply:
I combine category into type and change type to a object like this 

```ts
export const SearchCommandTypes = {
  PAGES: {
    description: i18n.translate('core.globalSearch.pages.description', { defaultMessage: 'Pages' }),
    alias: null,
  },
  SAVED_OBJECTS: {
    description: i18n.translate('core.globalSearch.assets.description', {
      defaultMessage: 'Assets',
    }),
    alias: SAVED_OBJECTS_SYMBOL,
  },
} as const;
```

Reply:
I think header_search_bar should better not maintain `page` or `save_objects`, because in the future, there might be `alerts` and `anomaly_detectors`, these are registered by plugins, which core should NOT be aware of, so the `description` and `alias` should be better registered by the plugin. What do you think?

Reply:
Let plugin to register it makes more sense. we could add `description` and `alias` to `GlobalSearchCommand` interface that will take precedence. And `SearchCommandTypes` serve as the builtin for pages and assets.